source/tests/system/nisync_driver_api_tests.cpp

```diff
@@ -1253,6 +1345,159 @@ TEST_F(NiSyncDriver6683Test, GetTimeReferenceNames_ReturnsSuccess)
+     &viStatusEnable);
+  EXPECT_TRUE(grpcStatusEnable.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatusEnable);
+
+  ViStatus viStatusDisable;
+  auto grpcStatusDisable = call_DisableTimeStampTrigger(
+     NISYNC_VAL_PFI1,
+     &viStatusDisable);
+  EXPECT_TRUE(grpcStatusDisable.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatusDisable);
+}
+
+TEST_F(NiSyncDriver6683Test, EnableTimeStampTriggerWithInvalidTerminal_ReturnsError)
+{
+  ViStatus viStatus;
+  auto grpcStatus = call_EnableTimeStampTrigger(
+     NISYNC_VAL_GND, // terminalName
+     NISYNC_VAL_EDGE_RISING, // activeEdge
+     &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(NISYNC_ERROR_TERMINAL_INVALID, viStatus);
+}
+
+TEST_F(NiSyncDriver6683Test, DisableTimeStampTriggerNotReserved_ReturnsError)
+{
+  ViStatus viStatus;
+  auto grpcStatus = call_DisableTimeStampTrigger(
+     NISYNC_VAL_PFI1, // terminalName
+     &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(NISYNC_ERROR_RSRC_NOT_RESERVED, viStatus);
+}
+
+TEST_F(NiSyncDriver6683Test, ReadTriggerTimeStamp_WhenNoTriggerOccurs_ReturnsTimeoutError)
+{
+  ViStatus viStatusEnable;
+  auto grpcStatusEnable = call_EnableTimeStampTrigger(
+     NISYNC_VAL_PFI1, // terminalName
+     NISYNC_VAL_EDGE_RISING, // activeEdge
+     &viStatusEnable);
+  EXPECT_TRUE(grpcStatusEnable.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatusEnable);
+
+  ViStatus viStatusRead;
+  ViReal64 timeout = 0.1;
+  ViUInt32 timeSeconds, timeNanoseconds;
+  ViUInt16 timeFractionalNanoseconds;
+  ViInt32 detectedEdge;
+  auto grpcStatusRead = call_ReadTriggerTimeStamp(
+     NISYNC_VAL_PFI1,
+     timeout,
+     &timeSeconds,
+     &timeNanoseconds,
+     &timeFractionalNanoseconds,
+     &detectedEdge,
+     &viStatusRead);
+  EXPECT_TRUE(grpcStatusRead.ok());
+  EXPECT_EQ(NISYNC_ERROR_DRIVER_TIMEOUT, viStatusRead);
+}
+
+TEST_F(NiSyncDriver6683Test, ReadMultipleTriggerTimeStamp_WhenNoTriggerOccurs_ReturnsTimeoutError)
+{
+  ViStatus viStatusEnable;
+  auto grpcStatusEnable = call_EnableTimeStampTrigger(
+     NISYNC_VAL_PFI1, // terminalName
+     NISYNC_VAL_EDGE_RISING, // activeEdge
+     &viStatusEnable);
+  EXPECT_TRUE(grpcStatusEnable.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatusEnable);
+
+  ViStatus viStatusRead;
+  ViReal64 timeout = 0.1;
+  ViUInt32 timestampsToRead = 1;
+  ViUInt32 timeSeconds, timeNanoseconds;
+  ViUInt16 timeFractionalNanoseconds;
+  ViInt32 detectedEdge;
+  ViUInt32 timestampsRead;
+  auto grpcStatusRead = call_ReadMultipleTriggerTimeStamp(
+     NISYNC_VAL_PFI1,
+     timeout,
+     timestampsToRead,
+     &timeSeconds,
+     &timeNanoseconds,
+     &timeFractionalNanoseconds,
+     &detectedEdge,
+     &timestampsRead,
+     &viStatusRead);
+  EXPECT_TRUE(grpcStatusRead.ok());
+  EXPECT_EQ(NISYNC_ERROR_DRIVER_TIMEOUT, viStatusRead);
+}
+
+TEST_F(NiSyncDriver6683Test, ReadMultipleTriggerTimeStamp_WhenFiveTriggersOccur_ReturnsFiveTimeStamps)
+{
+  auto terminal = NISYNC_VAL_PFI1;
+
+  // Ensure terminal level is low
```

Comment:
Maybe some of this setup can be factored out to 1 or more helpers to make the test easier to follow. Suggestions: `EnableTimestampingOnTerminal` & `CreateMultipleTimedEventPulses`. Then you can group the setup block together in this test to follow the given/when/then of the test name.

Reply:
I add methods `CreateFutureTimeEvent` and `EnableTimeStampTrigger` to wrapper the status handling of the `call_` flavors as the setup portion of the test requires the operations to be successful. I also cleaned up the `Create 5 pulses` section to be more readable. With the addition of `CreateFutureTimeEvent`, I decided not to wrap the for-loop in a method.